### DISCUSSION TITLE
How can I change the color of TextInput text based on the field value?

### DISCUSSION THREAD

**Discussion by LucaCVG** [filamentphp/filament#10004] [Help] [answered]
When viewing the record I would like to be able to change the color of the text of some fields in the image, based on their value. How i can? I'm a newbie : ) 

For example:
If Cost Variance >= 350000 -> Text color Green
If Cost Variance < 350000 ->  Text color Red

```
                    Forms\Components\TextInput::make('cost_variance')
                    ->label('Cost Variance (CV)')
                    ->hiddenOn('create')
                    //
                    //-> some code here to change text color based on field value
                    //
                    ->disabled()
                    ->prefix('€'),
```
![Schermata del 2023-12-01 17-39-31](https://github.com/filamentphp/filament/assets/55131838/b75e1e14-95ec-497b-876a-306ee92e34fe)

**Comment by leandrocfe** [comment, marked as answer]
You could use an infolist
https://filamentphp.com/docs/3.x/panels/resources/viewing-records#using-an-infolist-instead-of-a-disabled-form

and change the color using `->color()`

```php
TextEntry::make('cost_variance')
    ->color(fn ($state): string => intval($state) >= 350000 ? 'success' : 'danger');
```

**Reply by LucaCVG** [reply]
It worked the first time! Fantastic! Thank you very much!

![Schermata del 2023-12-01 21-29-37](https://github.com/filamentphp/filament/assets/55131838/979896c0-3181-4e09-bed4-3843c104d017)
